pkg/server/server.go

```diff
@@ -341,37 +369,12 @@ func (s *Server) SendVerificationLink(c *gin.Context, unverifiedEmail string) {
 		return
 	}
 
-	tgr := &jwtmanager.TokenGenerateRequest{
-		UserInfo:       updatedUserInfo,
-		AccessTokenExp: time.Minute * types.VerificationLinkExpirationTimeUnit,
-	}
-
-	tokenInfo, err := jwtmanager.GenerateAuthToken(s.accessGenerate, tgr, models.TokenVerify)
+	err = emailmanager.SendVerificationEmail(s.accessGenerate, updatedUserInfo)
 	if err != nil {
 		s.redirectError(c, err)
 		return
 	}
 
-	link := types.PortalURL + "/api/auth/v1/email?access=" + tokenInfo.Access
-
-	buf, err := generates.GetEmailBody(jwtUserCredentials.GetName(), link)
-	if err != nil {
-		log.Error("Error occurred while getting email body for user: " + jwtUserCredentials.GetUID() + "error: " + err.Error())
-		c.JSON(http.StatusInternalServerError, gin.H{
-			"error": err.Error(),
-		})
-		return
-	}
-
-	err = generates.SendEmail(unverifiedEmail, "Email Verification", buf.String())
-	if err != nil {
-		log.Error("Error occurred while sending email for user: " + jwtUserCredentials.GetUID() + "error: " + err.Error())
-		c.JSON(http.StatusInternalServerError, gin.H{
-			"error": err.Error(),
-		})
-		return
-	}
-
 	c.JSON(http.StatusOK, gin.H{
```

Comment:
Trying to understand this line of code. How/where the http status is fetched from as after refactor, verification email is sent from inside the email manager?

Reply:
Yes emailmanager is called here and if any error occurs it will return immediately. Else will reach this line of code and set status as `OK`